File: OshiLife/Data/Models/VenueSelection.swift
import CoreLocation
import Foundation

struct VenueSelection: Equatable, Sendable {
    let name: String
    let address: String
    let latitude: Double
    let longitude: Double

    init(name: String, address: String, coordinate: CLLocationCoordinate2D) {
        self.name = name
        self.address = address
        latitude = coordinate.latitude
        longitude = coordinate.longitude
    }

    init(name: String, address: String, latitude: Double, longitude: Double) {
        self.name = name
        self.address = address
        self.latitude = latitude
        self.longitude = longitude
    }
}

File: OshiLifeAndroid/lib/data/models/venue_selection.dart
/// Port of `OshiLife/Data/Models/VenueSelection.swift`.
class VenueSelection {
  const VenueSelection({
    required this.name,
    required this.address,
    required this.latitude,
    required this.longitude,
  });

  final String name;
  final String address;
  final double latitude;
  final double longitude;

  @override
  bool operator ==(Object other) {
    return other is VenueSelection &&
        other.name == name &&
        other.address == address &&
        other.latitude == latitude &&
        other.longitude == longitude;
  }

  @override
  int get hashCode => Object.hash(name, address, latitude, longitude);
}
